Source/Core/Core/HW/SI/SI_DeviceGBA.cpp

```diff
@@ -233,140 +214,138 @@ void GBASockServer::ClockSync()
   }
 }
 
-void GBASockServer::Send(const u8* si_buffer)
+bool GBASockServer::Connect()
 {
-  if (!m_client)
-    if (!GetAvailableSock(m_client))
-      return;
+  return m_client || (m_client = GetNextSock());
+}
 
-  for (size_t i = 0; i < m_send_data.size(); i++)
-    m_send_data[i] = si_buffer[i ^ 3];
+bool GBASockServer::IsConnected()
+{
+  return static_cast<bool>(m_client);
+}
 
-  m_cmd = (u8)m_send_data[0];
+void GBASockServer::Send(const u8* si_buffer)
+{
+  if (!Connect())
+    return;
 
-#ifdef _DEBUG
-  NOTICE_LOG(SERIALINTERFACE, "%01d cmd %02x [> %02x%02x%02x%02x]", m_device_number,
-             (u8)m_send_data[0], (u8)m_send_data[1], (u8)m_send_data[2], (u8)m_send_data[3],
-             (u8)m_send_data[4]);
-#endif
+  std::array<u8, 5> send_data;
+  for (size_t i = 0; i < send_data.size(); i++)
+    send_data[i] = si_buffer[i ^ 3];
+
+  u8 cmd = send_data[0];
+  if (cmd != CMD_STATUS)
+    m_booted = true;
 
   m_client->setBlocking(false);
   sf::Socket::Status status;
-  if (m_cmd == CMD_WRITE)
-    status = m_client->send(m_send_data.data(), m_send_data.size());
+  if (cmd == CMD_WRITE)
+    status = m_client->send(send_data.data(), send_data.size());
   else
-    status = m_client->send(m_send_data.data(), 1);
-
-  if (m_cmd != CMD_STATUS)
-    m_booted = true;
+    status = m_client->send(send_data.data(), 1);
```

Comment:
This has bugged me before, but do you happen to know why non-write uses a fixed size of 1 here?